pkg/operator/status/status_controller.go

```diff
@@ -113,22 +113,8 @@ func (c StatusSyncer) sync() error {
 	}
 	clusterOperatorObj := originalClusterOperatorObj.DeepCopy()
 
-	// if we are unmanaged, force everything to Unknown.
 	if detailedSpec.ManagementState == operatorv1.Unmanaged {
-		clusterOperatorObj.Status = configv1.ClusterOperatorStatus{}
-		configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorAvailable, Status: configv1.ConditionUnknown, Reason: "Unmanaged"})
-		configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorProgressing, Status: configv1.ConditionUnknown, Reason: "Unmanaged"})
-		configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorFailing, Status: configv1.ConditionUnknown, Reason: "Unmanaged"})
-		configv1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, configv1.ClusterOperatorStatusCondition{Type: configv1.OperatorUpgradeable, Status: configv1.ConditionUnknown, Reason: "Unmanaged"})
-
-		if equality.Semantic.DeepEqual(clusterOperatorObj, originalClusterOperatorObj) {
-			return nil
-		}
-		if _, updateErr := c.clusterOperatorClient.ClusterOperators().UpdateStatus(clusterOperatorObj); err != nil {
-			return updateErr
-		}
-		c.eventRecorder.Eventf("OperatorStatusChanged", "Status for operator %s changed: %s", c.clusterOperatorName, configv1helpers.GetStatusDiff(originalClusterOperatorObj.Status, clusterOperatorObj.Status))
-		return nil
+		c.eventRecorder.Eventf("OperatorStateInvalid", "Unmanaged state is not supported for operator %s", c.clusterOperatorName)
```

Comment:
@deads2k i hope this will be enough warning ?

Reply:
this isn't just used for control plane operators....library-go is a utility library that any operator can use.  So how is this safe?

Reply:
you can opt-in with your operator with `status.SetOperatorAlwaysManaged()`